File: Linkedlist/python/sumList.py
from __future__ import annotations
from dataclasses import dataclass
from typing import Optional


@dataclass
class Node:
    val: int
    next: Optional["Node"] = None


def sum_list(head: Optional[Node]) -> int:
    total = 0
    current = head
    while current is not None:
        total += current.val
        current = current.next
    return total


def sum_list_recursive(head: Optional[Node]) -> int:
    if head is None:
        return 0
    return head.val + sum_list_recursive(head.next)

File: Linkedlist/ts/sumList.ts
export class ListNode {
  constructor(public val: number, public next: ListNode | null = null) {}
}

export function sumList(head: ListNode | null): number {
  let sum = 0;
  let current = head;
  while (current !== null) {
    sum += current.val;
    current = current.next;
  }
  return sum;
}

export function sumListRecursive(head: ListNode | null): number {
  if (head === null) {
    return 0;
  }
  return head.val + sumListRecursive(head.next);
}
